### PR TITLE
HSEARCH-4041 + HSEARCH-4071 Remove deprecated interfaces/methods/properties introduced during the Alphas/Betas of 6.0.0

### PR DESCRIPTION
**Pull request by fax4ever**
https://hibernate.atlassian.net/browse/HSEARCH-4041
https://hibernate.atlassian.net/browse/HSEARCH-4071

~~Draft since I have to check some suppress deprecation removals.~~ let's try...

### CODE REVIEW

**Comment by yrodiere** [comment]
@fax4ever I launched a build with JDK8. Some of the deprecation warnings are there to make JDK8 happy, since it has bugs that make it issue more warning than necessary. Since deprecation warnings fail our build, that's a problem. We'll see...

File: engine/src/main/java/org/hibernate/search/engine/cfg/impl/EngineConfigurationUtils.java [hunk 1]
```diff
@@ -38,15 +38,12 @@ public static ConfigurationPropertySourceExtractor extractorForBackend(Optional<
...
 
-	@SuppressWarnings("deprecation")
 	public static ConfigurationPropertySourceExtractor extractorForIndex(
 			ConfigurationPropertySourceExtractor extractorForBackend, String indexName) {
 		return engineSource -> {
 			ConfigurationPropertySource backendSource = extractorForBackend.extract( engineSource );
 			return backendSource.withMask( BackendSettings.INDEXES ).withMask( indexName )
-					.withFallback( backendSource )
-					.withFallback( backendSource.withMask( BackendSettings.INDEX_DEFAULTS )
-							.onGet( log::deprecatedIndexDefaultsPrefix ) );
```
Comment (yrodiere) [inline engine/src/main/java/org/hibernate/search/engine/cfg/impl/EngineConfigurationUtils.java:54]: Looks like you're fixing https://hibernate.atlassian.net/browse/HSEARCH-4071 already. Fine by me, but maybe change the ticket key in the commit message?

File: engine/src/main/java/org/hibernate/search/engine/cfg/impl/EngineConfigurationUtils.java [hunk 2]
```diff
@@ -17,20 +17,9 @@
 	private EngineConfigurationUtils() {
 	}
 
-	public static ConfigurationPropertySourceExtractor extractorForBackend(Optional<String> backendNameOptional,
-			String defaultBackendName) {
+	public static ConfigurationPropertySourceExtractor extractorForBackend(Optional<String> backendNameOptional) {
 		if ( !backendNameOptional.isPresent() ) {
-			if ( defaultBackendName == null ) {
-				return engineSource -> engineSource.withMask( EngineSettings.Radicals.BACKEND );
```
Comment (yrodiere) [inline engine/src/main/java/org/hibernate/search/engine/cfg/impl/EngineConfigurationUtils.java:29]: Same for this commit: this looks like https://hibernate.atlassian.net/browse/HSEARCH-4071 material?

**Review by yrodiere — COMMENTED** [review]
Nice job! And it looks like you even addressed https://hibernate.atlassian.net/browse/HSEARCH-4071 : that's great.

I added a few comments, and I pushed a commit to your branch (for some reason GitHub won't let me add suggestions in my review...)

File: mapper/orm/src/main/java/org/hibernate/search/mapper/orm/scope/impl/SearchScopeImpl.java [hunk 3]
```diff
@@ -38,14 +41,19 @@ public SearchScopeImpl(HibernateOrmScopeMappingContext mappingContext,
...
-	@SuppressWarnings("deprecation")
-	public org.hibernate.search.mapper.orm.search.query.dsl.HibernateOrmSearchQuerySelectStep<E> search(HibernateOrmScopeSessionContext sessionContext) {
+	public SearchQuerySelectStep<
+			SearchQueryOptionsStep<?, E, SearchLoadingOptionsStep, ?, ?>,
+			EntityReference,
+			E,
+			SearchLoadingOptionsStep,
+			SearchProjectionFactory<EntityReference, E>,
+			SearchPredicateFactory
+			> search(HibernateOrmScopeSessionContext sessionContext) {
```
[the remark this answers is not in the document]
Comment (fax4ever) [reply mapper/orm/src/main/java/org/hibernate/search/mapper/orm/scope/impl/SearchScopeImpl.java]: Yes, makes sense to expose only what we really need. Thank you for the commit.

File: mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/mapping/building/impl/PojoRootIdentityMappingCollector.java [hunk 4]
```diff
@@ -78,19 +77,6 @@ void closeOnFailure() {
...
 	}
 
-	@Override
-	// In practice T and E are always the same, because the routing key bridge can only be applied at the root.
-	// Leaving this case temporarily, but we'll remove it along with RoutingKeyBridge.
-	@SuppressWarnings({"unchecked", "deprecation"})
-	public <T> void routingKeyBridge(BoundPojoModelPathTypeNode<T> modelPath,
-			org.hibernate.search.mapper.pojo.bridge.mapping.programmatic.RoutingKeyBinder binder) {
-		if ( routingBridge != null ) {
-			throw log.conflictingRoutingBridgeAndRoutingKeyBinder( routingBridge.getBridgeHolder().get(), binder );
```
[the remark this answers is not in the document]
Comment (fax4ever) [reply mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/mapping/building/impl/PojoRootIdentityMappingCollector.java:88]: Right. I missed it.

**Comment by fax4ever** [comment]
Thank you for the review. I think I've addressed you comments.

**Comment by yrodiere** [comment]
Looks good, thanks. Let's merge once CI is happy.

**Pull request merged**